boomonkeyboo/scratch_project
# server setup + getAllUsers + userLogIn + registerUser controllers done, corresponding routes added

**Issue Type**
☐ Bug
☑︎ Feature
☐ Tech Debt

**Description**

0. Set up node express server in server.ts
1. Created first route and first middleware (realized get all users)
2. Credit to Ryan, finished global error handler.
3. Created userLogIn, registerUser middlewares and corresponding routes

**Steps to Validate Feature**

0. Credit to Ryan, npm install
1. Run npm run dev to start back end
2. postman GET localhost:3001/api/getallusers, see response with some dummy data rows.
3. postman POST localhost:3001/api/registeruser, use {"fn":"dummyfn", "ln":"dummyln", "email":"dummyemail", "username":"dummyusername", "password":"dummypassword"} as req.body, check supabase for result. Try registering with same email/username, res should be a string describing the problem.
4. postman GET:localhost:3001/api/userlogin, use {"username":"[existing username]", "password":"[existing password]"} as req.body, res should be corresponding user. Try non-existent username or wrong passord, res should be a general information string followed by 00/01, see comment for details.
5. To test error handler, change 'userid' to 'useridd' in postgresModel.ts, repeat step 2, response should be a string describing getAllUsers failed.

**Previous behavior**
N/A

**Expected behavior**
Follow steps

**Screenshots & Videos** 
N/A 

**Additional context**  
Will communicate with front end developers and add more routes such as update and delete
